// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

// StressData is a message that includes all BigQuery data types.
message StressData {
    message KeyValuePair {
        optional string key = 1;
        optional string value = 2;
    }

    message SingleStruct {
        repeated int64 values = 1;
    }

    optional bool bool_value = 1;
    optional bytes bytes_value = 2;

    // days since unix epoch
    optional int32 date_value = 3;

    // datetime literal
    optional string datetime_value = 4;

    // wkt or geojson
    optional string geography_value = 5;
    optional int64 int64_value = 6;

    optional string numeric_value = 7;
    optional string bignumeric_value = 8;
    optional double float64_value = 9;
    optional string string_value = 10;

    // time literal
    optional string time_value = 11;

    // epoch microseconds
    optional int64 timestamp_value = 12;

    optional SingleStruct single_struct = 13;
    repeated KeyValuePair pairs = 14;
}
